backend/app/api/endpoints/knowledge_import.py

```
@@ -0,0 +1,148 @@
+
+
+@router.post(
+    "/{knowledge_base_id}/external-imports",
+    response_model=ExternalKnowledgeImportResponse,
+    status_code=status.HTTP_201_CREATED,
+)
+async def import_external_content(
+    knowledge_base_id: int,
+    data: ExternalKnowledgeImportRequest,
+    background_tasks: BackgroundTasks,
+    auth_context: AuthContext = Depends(security.get_auth_context),
```

Ruff (0.14.13)

Do not perform function call `Depends` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)